### PR TITLE
Package cb-check.0.1

### PR DESCRIPTION
**Pull request by ElectreAAS**
### `cb-check.0.1`
Json schema checker for current-bench

---
* Homepage: https://github.com/ocurrent/current-bench
* Source repo: git+https://github.com/ocurrent/current-bench.git
* Bug tracker: https://github.com/ocurrent/current-bench/issues

---
:camel: Pull-request generated by opam-publish v2.2.0

### CODE REVIEW

File: packages/cb-check/cb-check.0.1/opam [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+synopsis: "Json schema checker for current-bench"
+maintainer: ["Ambre Austen Suhamy <ambre@tarides.com>"]
+authors: ["Ambre Austen Suhamy <ambre@tarides.com>"]
+homepage: "https://github.com/ocurrent/current-bench"
+bug-reports: "https://github.com/ocurrent/current-bench/issues"
+depends: [
+  "dune" {>= "3.7"}
+  "ocaml" {>= "4.13.0"}
+  "yojson"
+  "odoc" {with-doc}
```
Comment (mseri) [inline packages/cb-check/cb-check.0.1/opam:12]:
```suggestion
  "uri" {with-test}
  "odoc" {with-doc}
```

File: packages/cb-check/cb-check.0.1/opam [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# This file is generated by dune, edit dune-project instead
+opam-version: "2.0"
+synopsis: "Json schema checker for current-bench"
+maintainer: ["Ambre Austen Suhamy <ambre@tarides.com>"]
+authors: ["Ambre Austen Suhamy <ambre@tarides.com>"]
```
Comment (mseri) [inline packages/cb-check/cb-check.0.1/opam:5]: The license field is missing, can you add it?

**Comment by avsm** [comment]
@ElectreAAS once you decide on a license, you can push directly to this PR to add the license field to the `opam` file.

File: packages/cb-check/cb-check.0.1/opam (hunk 2 above)
Comment (ElectreAAS) [reply packages/cb-check/cb-check.0.1/opam:5]: Should be done now. Thanks for checking :)

**Comment by ElectreAAS** [comment]
License added!

File: packages/cb-check/cb-check.0.1/opam [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+license: "Apache-2.0"
+homepage: "https://github.com/ocurrent/current-bench"
+bug-reports: "https://github.com/ocurrent/current-bench/issues"
+depends: [
+  "dune" {>= "3.7"}
+  "ocaml" {>= "4.13.0"}
+  "yojson"
+  "uri" {with-test}
+  "odoc" {with-doc}
```
Comment (mseri) [inline packages/cb-check/cb-check.0.1/opam:14]:
```suggestion
  "ppx_deriving_yojson" {with-test}
  "timere-parse" {with-test}
  "odoc" {with-doc}
```

File: packages/cb-check/cb-check.0.1/opam [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+bug-reports: "https://github.com/ocurrent/current-bench/issues"
+depends: [
+  "dune" {>= "3.7"}
+  "ocaml" {>= "4.13.0"}
+  "yojson"
+  "uri" {with-test}
+  "ppx_deriving_yojson" {with-test}
+  "timere-parse" {with-test}
+  "rresult" {with-test}
+  "odoc" {with-doc}
```
Comment (mseri) [inline packages/cb-check/cb-check.0.1/opam:17]:
```suggestion
  "prometheus" {with-test}
  "odoc" {with-doc}
```
Due to
```
#=== ERROR while compiling cb-check.0.1 =======================================#
# context              2.2.0~alpha~dev | linux/arm32 | ocaml-base-compiler.4.14.1 | pinned(https://github.com/ocurrent/current-bench/archive/refs/tags/0.1.zip)
# path                 ~/.opam/4.14/.opam-switch/build/cb-check.0.1
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p cb-check -j 79 @install @runtest
# exit-code            1
# env-file             ~/.opam/log/cb-check-7-c1f93b.env
# output-file          ~/.opam/log/cb-check-7-c1f93b.out
### output ###
# File "pipeline/tests/dune", line 20, characters 2-12:
# 20 |   prometheus
#        ^^^^^^^^^^
# Error: Library "prometheus" not found.
# -> required by _build/default/pipeline/tests/test.exe
# -> required by alias pipeline/tests/runtest in pipeline/tests/dune:2
```

File: packages/cb-check/cb-check.0.1/opam [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+depends: [
+  "dune" {>= "3.7"}
+  "ocaml" {>= "4.13.0"}
+  "yojson"
+  "uri" {with-test}
+  "ppx_deriving_yojson" {with-test}
+  "timere-parse" {with-test}
+  "rresult" {with-test}
+  "prometheus" {with-test}
+  "odoc" {with-doc}
```
Comment (mseri) [inline packages/cb-check/cb-check.0.1/opam:18]:
```suggestion
  "logs" {with-test & >= "0.5.0"}
  "odoc" {with-doc}
```
due to
```

#=== ERROR while compiling cb-check.0.1 =======================================#
# context              2.2.0~alpha~dev | linux/arm32 | ocaml-base-compiler.4.14.1 | pinned(https://github.com/ocurrent/current-bench/archive/refs/tags/0.1.zip)
# path                 ~/.opam/4.14/.opam-switch/build/cb-check.0.1
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p cb-check -j 79 @install @runtest
# exit-code            1
# env-file             ~/.opam/log/cb-check-7-b569f6.env
# output-file          ~/.opam/log/cb-check-7-b569f6.out
### output ###
# File "pipeline/tests/dune", line 19, characters 2-10:
# 19 |   logs.fmt
#        ^^^^^^^^
# Error: Library "logs.fmt" not found.
# -> required by _build/default/pipeline/tests/test.exe
# -> required by alias pipeline/tests/runtest in pipeline/tests/dune:2
```w

**Comment by mseri** [comment]
@ElectreAAS there are a number of missing test dependencies, can you have a look and update the opam file with them?

**Comment by mseri** [comment]
The problem is that all the tests are not associated to a specific package, so they will be built for all packages in the repository (including this one). To avoid this, the test entries should contain a `(package pkgname)` entry. 

Since this package has no tests, an alternative would be to disable tests altogether

File: packages/cb-check/cb-check.0.1/opam [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  ["dune" "subst"] {dev}
+  [
+    "dune"
+    "build"
+    "-p"
+    name
+    "-j"
+    jobs
+    "@install"
+    "@runtest" {with-test}
```
Comment (mseri) [inline packages/cb-check/cb-check.0.1/opam:31]:
```suggestion
```

if you decide to go with this, please remove the test dependencies added above

Comment (ElectreAAS) [reply packages/cb-check/cb-check.0.1/opam]: Done!

**Comment by mseri** [comment]
Thanks

**Pull request merged**